File: fib.java
/*
 * javac fib.java
 * java fib
 */

public class fib
{
  private static long fib(long n)
  {
    if (n < 3)
      return 1;
    return fib(n-1) + fib(n-2);
  }

  public static void main(String[] args)
  {
    System.out.println(fib(44));
    System.out.println(fib(45));
    System.out.println(fib(46));
  }
}

File: fib.hs
module Main( main ) where

fib :: Int -> Int
fib x = if x < 3 then 1
        else (fib(x - 1) + fib(x - 2))

main = do
  print("fib(44) = " ++ show (fib 44))
  print("fib(45) = " ++ show (fib 45))
  print("fib(46) = " ++ show (fib 46))
  putStr "\n"
